overlord/overlord.go

```diff
@@ -209,7 +215,45 @@ func (o *Overlord) addManager(mgr StateManager) {
 	o.stateEng.AddManager(mgr)
 }
 
-func loadState(backend state.Backend, restartHandler restart.Handler) (*state.State, error) {
+func initStateFileLock() (*osutil.FileLock, error) {
+	lockFilePath := dirs.SnapStateLockFile
+	if err := os.MkdirAll(filepath.Dir(lockFilePath), 0755); err != nil {
+		return nil, err
+	}
+
+	return osutil.NewFileLockWithMode(lockFilePath, 0644)
+}
+
+func lockWithTimeout(l *osutil.FileLock, timeout time.Duration) error {
+	startTime := time.Now()
+	const retryInterval = time.Second
+	for {
+		err := l.TryLock()
+		if err != osutil.ErrAlreadyLocked {
```

Comment:
In general explicit is better than explicit, and it is true that in its current form the code requires the reader to stop and think. I opted for adding a comment instead, because I see a small risk that someone in the future will "optimize" your suggested code and revert it to the compact form (without adding a comment :-) ).

Reply:
thanks the comment is sufficient